src/components/tables/AccessGrants/index.tsx

```diff
@@ -1,67 +1,60 @@
 import { Box } from '@mui/material';
+import { getGrantsForEverything } from 'api/combinedGrantsExt';
 import Rows, { tableColumns } from 'components/tables/AccessGrants/Rows';
-import EntityTable, { getPagination } from 'components/tables/EntityTable';
-import { useQuery } from 'hooks/supabase-swr';
-import { useState } from 'react';
-import { defaultTableFilter, TABLES } from 'services/supabase';
+import EntityTable from 'components/tables/EntityTable';
+import { useMemo } from 'react';
 import { SelectTableStoreNames } from 'stores/names';
-import { SortDirection } from 'types';
-
-// const queryColumns = ['id', 'spec_type', 'catalog_name', 'updated_at'];
+import TableHydrator from 'stores/Tables/Hydrator';
+import useTableState from '../hooks';
 
 function AccessGrantsTable() {
-    const rowsPerPage = 10;
-    const [pagination, setPagination] = useState<{ from: number; to: number }>(
-        getPagination(0, rowsPerPage)
-    );
-    const [searchQuery, setSearchQuery] = useState<string | null>(null);
-    const [sortDirection, setSortDirection] = useState<SortDirection>('asc');
-    const [columnToSort, setColumnToSort] = useState<any>('user_full_name');
+    const {
+        pagination,
+        setPagination,
+        searchQuery,
+        setSearchQuery,
+        sortDirection,
+        setSortDirection,
+        columnToSort,
+        setColumnToSort,
+    } = useTableState('user_full_name');
 
-    const rolesQuery = useQuery<any>(
-        TABLES.COMBINED_GRANTS_EXT,
-        {
-            columns: `id, subject_role, object_role, capability, user_avatar_url, user_full_name, user_email, updated_at`,
-            count: 'exact',
-            filter: (query) => {
-                return defaultTableFilter<any>(
-                    query,
-                    ['user_full_name', 'subject_role', 'object_role'],
-                    searchQuery,
-                    [
-                        {
-                            col: columnToSort,
-                            direction: sortDirection,
-                        },
-                    ],
-                    pagination
-                );
+    const query = useMemo(() => {
+        return getGrantsForEverything(pagination, searchQuery, [
+            {
+                col: columnToSort,
+                direction: sortDirection,
             },
-        },
-        [pagination, searchQuery, columnToSort, sortDirection]
-    );
+        ]);
+    }, [columnToSort, pagination, searchQuery, sortDirection]);
```

Comment:
New approach for queries... just create a fetched based on the API stuff and memoize it.